pkg/deploy/service.go

```diff
@@ -307,3 +283,103 @@ func contains(s []string, searchterm string) bool {
 	i := sort.SearchStrings(s, searchterm)
 	return i < len(s) && s[i] == searchterm
 }
+
+func buildStrategy(configStrategies map[string]model.Strategy, strategyName string) (*de.PipelinePipelineStrategy, error) {
+	strategy := configStrategies[strategyName]
+	if strategy.Canary != nil {
+		steps, err := CreateDeploymentCanaryStep(strategy)
+		if err != nil {
+			return nil, err
+		}
+		return &de.PipelinePipelineStrategy{
+			Canary: &de.KubernetesV2CanaryStrategy{
+				Steps: steps,
+			},
+		}, nil
+	} else if strategy.BlueGreen != nil {
+		redirectTrafficAfter, err := createRedirectTrafficAfter(strategy)
+		if err != nil {
+			return nil, err
+		}
+		shutdownOldVersionAfter, err := createShutdownOldVersionAfter(strategy)
+		if err != nil {
+			return nil, err
+		}
+		return &de.PipelinePipelineStrategy{
+			BlueGreen: &de.KubernetesV2BlueGreenStrategy{
+				RedirectTrafficAfter: redirectTrafficAfter,
+				ShutdownOldVersionAfter: shutdownOldVersionAfter,
+				ActiveService: strategy.BlueGreen.ActiveService,
+				PreviewService: strategy.BlueGreen.PreviewService,
+				ActiveUrl: &strategy.BlueGreen.ActiveRootUrl,
+				PreviewUrl: &strategy.BlueGreen.PreviewRootUrl,
+			},
+		}, nil
+	}
+
+	return nil, fmt.Errorf("%s is not a valid strategy; define canary or blue-green strategy", strategyName)
+}
+
+func createRedirectTrafficAfter(strategy model.Strategy) (*de.KubernetesV2RedirectTrafficAfter, error) {
+	steps := make([]de.KubernetesV2BlueGreenStep, 0, len(*strategy.BlueGreen.RedirectTrafficAfter.Steps))
```

Comment:
```suggestion
	var steps []de.KubernetesV2BlueGreenStep
```
I don't know if there's any reason to initialize the slice as you've done unless there are performance concerns.